--- a/pw_kernel/kernel/tests/BUILD.bazel
+++ b/pw_kernel/kernel/tests/BUILD.bazel
@@ -25,6 +25,8 @@
     ],
     crate_features = select({
         "@platforms//cpu:armv8-m": ["arch_arm_cortex_m"],
+        "@platforms//cpu:armv7-m": ["arch_arm_cortex_m"],
+        "@platforms//cpu:armv7e-m": ["arch_arm_cortex_m"],
         "@platforms//cpu:riscv32": ["arch_riscv"],
         "//conditions:default": [],
     }),
@@ -33,6 +35,8 @@
     tags = ["kernel"],
     target_compatible_with = select({
         "@platforms//cpu:armv8-m": [],
+        "@platforms//cpu:armv7-m": [],
+        "@platforms//cpu:armv7e-m": [],
         "@platforms//cpu:riscv32": [],
         "//conditions:default": ["@platforms//:incompatible"],
     }),
@@ -42,6 +46,8 @@
         "//pw_kernel/lib/magic_values",
     ] + KERNEL_TEST_DEPS + select({
         "@platforms//cpu:armv8-m": ["//pw_kernel/arch/arm_cortex_m:arch_arm_cortex_m"],
+        "@platforms//cpu:armv7-m": ["//pw_kernel/arch/arm_cortex_m:arch_arm_cortex_m"],
+        "@platforms//cpu:armv7e-m": ["//pw_kernel/arch/arm_cortex_m:arch_arm_cortex_m"],
         "@platforms//cpu:riscv32": ["//pw_kernel/arch/riscv:arch_riscv"],
         "//conditions:default": [],
     }),
